code-challenges.test.js

```diff
@@ -0,0 +1,135 @@
+// ASSESSMENT 4: JavaScript Coding Practical Questions with Jest
+
+// Please read all questions thoroughly
+// Pseudo coding is REQUIRED
+// If you get stuck, please leave comments to help us understand your thought process
+
+// Use test driven development to complete the following questions
+// Add appropriate dependencies: $ yarn add jest
+
+// Reminder: The test will call your function
+// Run the file with the following command: $ yarn jest
+
+// --------------------1) Create a function that takes in an array, removes the first item from the array and shuffles the remaining content.
+
+// a) Create a test with an expect statement using the variable provided.
+// HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
+
+describe("removeFirstItem", () => {
+  it("removes the first item from the array and shuffles the remaining content", () => {
+const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+  
+  const colors2 = [
+  "chartreuse",
+  "indigo",
+  "periwinkle",
+  "ochre",
+  "aquamarine",
+  "saffron"
+]
+// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+expect(removeFirstItem(colors1)).toEqual(expect.arrayContaining(colors1))
+expect(removeFirstItem(colors2)).toEqual(expect.arrayContaining(colors2))
```

Comment:
Your function is essentially doing what the prompt is asking, but your test is not set up correctly. You are telling it to expect the output to be an array containing all the elements in the original array. However, that's not what you are expecting, as you should be expecting an array containing all the elements in the original array except the first one, since the prompt is asking you to remove it. This test currently passes only because you use `.shift()`, which is a mutator. So by the time the test gets to checking the output, the original array has been mutated already, so it's performing a false check. To have a proper test, you should have the expected outputs that the prompt gave you in the argument for arrayContaining instead of the variable names.